File: Mnema.Frontend/Web/src/app/_models/page.ts
import {FormControlDefinition} from "../generic-form/form";
import {MetadataBag} from "@mnema/_models/search";

export type Page = {
  id: string;
  sortValue: number;
  title: string;
  icon: string;
  provider: Provider;
  modifiers?: FormControlDefinition[];
  metadata?: FormControlDefinition[],
  customRootDir: string;
  defaultOptions: MetadataBag;
}

export enum Provider {
  NYAA = 0,
  MANGADEX = 1,
  WEBTOON = 2,
  DYNASTY = 3,
  BATO = 4,
  WEEBDEX = 5,
  COMIX = 6,
  KAGANE = 7,
  Madokami = 8,
  AthreaScans = 9,
}

export const InUseProviders: Provider[] = [
  Provider.NYAA,
  Provider.MANGADEX,
  Provider.WEBTOON,
  Provider.DYNASTY,
  Provider.Madokami,
  Provider.AthreaScans,
]

File: Mnema.Models/Entities/Page.cs
using System;
using Mnema.Common;
using Mnema.Models.Entities.Interfaces;
using Mnema.Models.Enums;

namespace Mnema.Models.Entities;

public class Page: IDatabaseEntity
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public string Icon { get; set; } = string.Empty;
    public required int SortValue { get; set; }
    public required Provider Provider { get; set; }
    public string CustomRootDir { get; set; } = string.Empty;
    public MetadataBag DefaultOptions { get; set; } = new();
}
